sys/ufs/ffs/ffs_alloc.c

```diff
@@ -3221,10 +3237,31 @@ sysctl_ffs_fsck(SYSCTL_HANDLER_ARGS)
 	cap_rights_t rights;
 	int filetype, error;
 
-	if (req->newptr == NULL || req->newlen > sizeof(cmd))
+	if (req->newptr == NULL)
 		return (EBADRPC);
-	if ((error = SYSCTL_IN(req, &cmd, sizeof(cmd))) != 0)
-		return (error);
+	switch (req->newlen) {
+	case sizeof(cmd):
+		if ((error = SYSCTL_IN(req, &cmd, sizeof(cmd))) != 0)
+			return (error);
+		break;
+#ifdef COMPAT_FREEBSD64
```

Comment:
Hmm, is there an SCTL_64 flag?  We should probably use that instead of size checks to avoid a purecap process being able to forge a pointer that the kernel will read.

Reply:
There's no SCTL_64, I think I need to check SV_CHERI.

Reply:
FWIW, the kernel can't be tricked into creating a valid pointer because the thread's DDC is used and it's NULL.

Reply:
(and checking SV_CHERI needs to be done with care given kernel_sysctl and db_sysctl)

Reply:
To be clear, I would suggest requiring the size to be the freebsd64 size for SCTL_MASK64 and the native size otherwise rather than switching on the size.  True about the null DDC for purecap, though I'd like to avoid depending on that (a purecap process could have set ddc to something).

Reply:
I think I've correctly switched to using SCTL_MASK64.